appscripts/canalytics.gs

```diff
@@ -145,40 +153,212 @@ function scoreBreakdownSummaryHeader(){
+          endgameScore +=10
+        }
+        summaryObjects[scoutingItem['Team Number']]['Avg_Endgame_Score'] += endgameScore
+      }
+
+      if ( summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] == "none" ) {
+        if (scoutingItem['Pieces Scored on Mid/High'] == "BOTH" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CUBES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "cubes"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CONES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "cones"
+          }
+      }
+      if ( summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] == "cones" ) {
+        if (scoutingItem['Pieces Scored on Mid/High'] == "BOTH" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CUBES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+      }
+      if ( summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] == "cubes" ) {
+        if (scoutingItem['Pieces Scored on Mid/High'] == "BOTH" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CONES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+      }
+   })  
+  Object.keys(summaryObjects).forEach( k => {
+    if ( summaryObjects[k]['Matches_Played'] > 0 ) {
+      summaryObjects[k]['Avg_Auton_Score'] = summaryObjects[k]['Avg_Auton_Score']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Auton_GPs'] = summaryObjects[k]['Avg_Auton_GPs']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Teleop_Score (no links)'] = summaryObjects[k]['Avg_Teleop_Score (no links)']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Teleop_GPs'] = summaryObjects[k]['Avg_Teleop_GPs']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Endgame_Score'] = summaryObjects[k]['Avg_Endgame_Score']/summaryObjects[k]['Matches_Played']
+    }
+  })
+
+  return summaryObjects
+}
+
+function specificScoutingSummaryHeader(){
+  return ["Auton Score", "Teleop Score", "Endgame Score", "Total Score"]
+}
+
+function specificScoutingSummary(specificTeamObjects, specificScoutingObjects){
+  Object.keys(specificTeamObjects).forEach( k => {
+      specificTeamObjects[k]['Auton Score']=0;
+      specificTeamObjects[k]['Teleop Score']=0;
+      specificTeamObjects[k]['Endgame Score']=0;
+      specificTeamObjects[k]['Total Score']=0;
+   })
+  Object.keys(specificScoutingObjects).forEach( k => {
+      specificScoutingItem = specificScoutingObjects[k]
+      var autonScore = 0  
```

Comment:
You could initialize some data here, maybe?  
specificScoutingItem['Auton Score'] = 0
specificScoutingItem['Teleop Score'] = 0
...